src/csharp/Grpc.Core/Channel.cs

```diff
@@ -84,6 +85,13 @@ public Channel(string target, ChannelCredentials credentials, IEnumerable<Channe
                     this.handle = ChannelSafeHandle.CreateInsecure(target, nativeChannelArgs);
                 }
             }
+
+            if (credentials.HasGcHandle)
+            {
+                var gcHandle = credentials.GCHandle;
+                cleanUpCredentials = () => gcHandle.Free();
```

Comment:
The reason for using an `Action` instead of just a `GCHandle` field is simply to avoid creating and `GCHandle` when it is not needed. It would not have a noticeable impact but just thought that this approach was a tiny bit cleaner.

Reply:
No longer relevant on latest verison.